lib/db/mysql/scripts/purge-expired-tokens.sql

```diff
@@ -0,0 +1,43 @@
+DROP PROCEDURE IF EXISTS PURGE_EXPIRED_TOKENS;
+
+DELIMITER $$
+# Create a procedure that will delete a total of `roundCount` * `limitDeletePerRound` items, every `delaySeconds`.
+# Skip tokens that belong clientId
+CREATE PROCEDURE PURGE_EXPIRED_TOKENS(IN roundCount INT, IN limitDeletePerRound INT, IN delaySeconds INT, IN ignoreClientId CHAR(10))
+
+BEGIN
+  DECLARE startIndex INT DEFAULT 1 ;
+  DECLARE endCount INT DEFAULT roundCount ;
+  DECLARE expiredItemCount INT DEFAULT 0;
+
+  PURGE_LOOP: LOOP
+
+	# Exit loop if no expired items found
+    SET expiredItemCount = (SELECT COUNT(*) FROM tokens WHERE expiresAt < NOW() AND UNHEX(clientId) != ignoreClientId ORDER BY expiresAt);
```

Comment:
I wonder whether we need to do this as a separate explicit `count`, or whether we can ask MySQL how many rows it successfully deleted in the query below, and exit if it didn't find any rows to delete.